# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/websocket/WebSocketConnectionChild.cpp
# Commit: a2372b6285d7
# Full Hash: a2372b6285d7871f7eb6e914583b1113da2038a1
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-09-11 15:10:42
# Description:
#   Bug 1663718 - Don't put too much data in buffer when the data can't be written to socket r=dragana
#   
#   The main reason we used too much memory is that we ignore the `NS_BASE_STREAM_WOULD_BLOCK` returned from socket output stream.
#   When the output stream is blocked, all the data is stored in the output queue and make the memory usage high.
#   
# ==============================================================================

diff -r 433c4ac7d65a -r a2372b6285d7 netwerk/protocol/websocket/WebSocketConnectionChild.cpp
--- a/netwerk/protocol/websocket/WebSocketConnectionChild.cpp	Fri Sep 11 14:13:46 2020 +0300
+++ b/netwerk/protocol/websocket/WebSocketConnectionChild.cpp	Fri Sep 11 08:44:07 2020 +0000
@@ -174,6 +174,12 @@
   return NS_OK;
 }
 
+NS_IMETHODIMP
+WebSocketConnectionChild::OnReadyToSendData() {
+  // TODO: implement flow control between parent and socket process.
+  return NS_OK;
+}
+
 void WebSocketConnectionChild::ActorDestroy(ActorDestroyReason aWhy) {
   LOG(("WebSocketConnectionChild::ActorDestroy %p\n", this));
   if (mConnection) {